server/chat-commands.js

```diff
@@ -3796,6 +3796,49 @@ const commands = {
 	},
 	importinputloghelp: [`/importinputlog [inputlog] - Starts a battle with a given inputlog. Requires: + % @ & ~`],
 
+	acceptdraw: 'accepttie',
+	offerdraw: 'offertie',
+	accepttie: 'offertie',
+	offertie(target, room, user, connection, cmd) {
+		const battle = room.battle;
+		if (!battle) return this.errorReply("Must be in a battle room.");
+		if (!Config.allowrequestingties) {
+			return this.errorReply("This server does not allow offering ties.");
+		}
+		if (!this.can('roomvoice', null, room)) return;
+		if (cmd === 'accepttie' && !battle.players.some(player => player.wantsTie)) {
+			return this.errorReply("No tie offer available. It might have been automatically withdrawn as the current turn started.");
+		}
+		const playerIds = Object.keys(battle.playerTable);
```

Comment:
This is unnecessary; `players` exists. Unrelatedly, caching the current player is probably useful here.

```suggestion
		const player = battle.playerTable[user.userid];
```